lib/matplotlib/axes/_axes.py

```diff
@@ -1791,16 +1791,13 @@ def loglog(self, *args, **kwargs):
         **kwargs
             All parameters supported by `.plot`.
         """
-        dx = {k: kwargs.pop(k) for k in ['basex', 'subsx', 'nonposx']
+        dx = {k[:-1]: kwargs.pop(k) for k in ['basex', 'subsx', 'nonposx']
```

Comment:
Sorry, why are you excluding the last element here?

Reply:
Because the kwarg name for semilogx/y is still "basex"; this PR doesn't change that.  Specifically, while I could change the kwarg name in semilogx to "base" (with the same deprecation), this doesn't work for loglog(), which needs a separate basex/basey (although it would also be reasonable to deprecate supporting directly setting two different bases for x and y in loglog() and require the user to call set_x/yscale("log", base=...) for that specific case.  I think we just need to decide on the best API change here.

Reply:
OK.  Duh, I see now, but this might be a rare case where not using comprehension might be a lot clearer: 
```python
for k in ['basex', 'subex', 'nonposx']:
  dx{k[:-1]} = kwargs.pop(k)
```

Reply:
I agree that the side-effect of popping kwargs in the comprehension may not be nicest, and can change that, but before doing so, do you have comments on the API changes suggested above?

Reply:
I agree that most folks won't use different `basex`, `basey`.  OTOH there is call to use different `subs` and `nonpos` isn't there?

Reply:
People can still use `set_xscale("log", subs=...)` in that case.  Note that in the (common?) case of using the same subs on x and y, the proposed new API is actually nicer (`loglog(..., subs=...)` instead of repeating yourself with `subs(..., subsx=..., subsy=...)`).
I have kind of changed my mind compared to when I first wrote the PR; I now think that having the same kwargs in `loglog`/`semilogx/y` as in `set_scale("log", ...)` would actually be quite nice (whereas I was previously more ok with keeping basex/basey on loglog/semilog).

Reply:
I'd be onboard with this, but we should prob be prepared to not do this if the deprecation warning freaks too many people out.  

Other comments?  @timhoffm @ImportanceOfBeingErnest @efiring?

Reply:
basex, basey are poor names anyway; I would be happy to see them go.  This sort of regularization of the API will inevitably cause some pain, but I think we should do it.